content/blog/2015-06-17-practical-anomaly-detection.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Practical Anomaly Detection
+created_at: 2015-06-17
+kind: article
+author_name: Brian Brazil
+---
+
+In his [Open Letter To Monitoring/Metrics/Alerting Companies](http://www.kitchensoap.com/2015/05/01/openlettertomonitoringproducts/),
+John Allspaw asserts that attempting "to detect anomalies perfectly, at the right time, is not possible".
+
+I have seen more than one attempt by talented engineers to build a system to
+automatically detect and diagnose problems based on time series data. While it
+is certainly possible to get a demonstration working, the data always turned
+out to be too noisy to make this approach work for anything but the simplest of
+real-world systems.
+
+All hope is not lost though. There are some common anomalies which you can
+detect and deal with. The Prometheus [query
+language](../../../../../docs/querying/basics/) gives you the tools to discover
+these while avoiding false positives.
+
+## Building a query
+
+A common problem is a small number of servers within a services not
+performing as well as the rest, such as by having increased latency.  While
+the issue may not impact your overall SLA, it is worth having a human look when
+they have a chance.
+
+Let us say that we have `instance:latency_seconds:mean5m` representing the
+average query latency for each instance of a service, calculated from a Summary
```

Comment:
Not sure how well-known the meaning of summary is – maybe link to http://prometheus.io/docs/concepts/metric_types/#summary